### PR TITLE
feat(webpack-cli): add arg postprocessing

### PR DESCRIPTION
**Pull request by ylemkimon**
**What kind of change does this PR introduce?**
Feature

**Did you add tests for your changes?**
Yes

**If relevant, did you update the documentation?**
N/A

**Summary**
- Added `arg.processor` for post-processing options, e.g., migration, deprecation, renaming, nesting, etc.
  - Examples: https://github.com/webpack/webpack-dev-server/pull/2857/files#diff-6639cb4cd2a1457aacae2c81ab3040c3c3fd959572d96019fffc5294ca34411e
- Removed `createConfig` in favor of `webpack-dev-server/bin/cli-flags`

**Does this PR introduce a breaking change?**
`@webpack-cli/server/lib/createConfig` is removed.

**Other information**

### CODE REVIEW

**Review by evilebottnawi — APPROVED** [review]

File: packages/webpack-cli/__tests__/arg-parser.test.js [hunk 1]
```diff
@@ -96,6 +96,16 @@ const basicOptions = [
...
         description: 'multi flag',
     },
+    {
+        name: 'processor-flag',
+        usage: '--processor-flag',
+        type: Boolean,
+        description: 'flag with processor',
+        processor(opts) {
+            opts.processed = opts.processorFlag;
+            delete opts.processorFlag;
```
Comment (snitin315) [inline packages/webpack-cli/__tests__/arg-parser.test.js:106]: Can we do ?

```suggestion
        processor: (opts) => {
            opts.processed = opts.processorFlag;
            delete opts.processorFlag;
```

Comment (evilebottnawi) [reply packages/webpack-cli/__tests__/arg-parser.test.js:106]: Why?

Comment (snitin315) [reply packages/webpack-cli/__tests__/arg-parser.test.js:106]: just to match like other properties.

Comment (evilebottnawi) [reply packages/webpack-cli/__tests__/arg-parser.test.js:106]: ArrowFunctionExpression and MethodDefinition is different thing and can used for different purposes

**Review by snitin315 — APPROVED** [review]

**Pull request merged**